app/src/main/java/com/github/libretube/api/NewPipeMediaServiceRepository.kt

```diff
@@ -319,14 +321,50 @@ class NewPipeMediaServiceRepository : MediaServiceRepository {
         )
     }
 
+    @OptIn(ExperimentalStdlibApi::class)
     override suspend fun getSegments(
         videoId: String,
-        category: String,
-        actionType: String?
-    ): SegmentData = SegmentData()
+        category: List<String>,
+        actionType: List<String>?
+    ): SegmentData {
+        // use hashed video id for privacy
+        // https://wiki.sponsor.ajay.app/w/API_Docs#GET_/api/skipSegments/:sha256HashPrefix
+        val hashedId = MessageDigest.getInstance("SHA-256")
+            .digest(videoId.toByteArray())
+            .toHexString()
+
+        return RetrofitInstance.externalApi.getSegments(
+            hashedId.substring(0..4),
```

Comment:
`substring(0..4)` would be `hashedId[0]...hashedId[4]` (5 chars), while below you used `substring(0, 4)` which excludes `hashedId[4]`. Probably a typo?

Reply:
Just me being used to Rust Ranges :)